server/src/main/java/org/jboss/as/server/operations/AbstractInstallationReporter.java

```diff
@@ -0,0 +1,303 @@
+        jvm.get(JVM_VENDOR).set(getProperty("java.vm.vendor"));
+        jvm.get(JVM_HOME).set(getProperty("java.home"));
+        return jvm;
+    }
+
+    /**
+     * Create a ModelNode representing the CPU the instance is running on.
+     *
+     * @return a ModelNode representing the CPU the instance is running on.
+     * @throws OperationFailedException
+     */
+    private ModelNode createCPUNode() throws OperationFailedException {
+        ModelNode cpu = new ModelNode().setEmptyObject();
+        cpu.get(ARCH).set(getProperty("os.arch"));
+        cpu.get(AVAILABLE_PROCESSORS).set(Runtime.getRuntime().availableProcessors());
+        return cpu;
+    }
+
+    /**
+     * Create a ModelNode representing the instance product and the plateform it is running on.
+     *
+     * @param context the operation context.
+     * @param environment the current runtime environment.
+     * @param patchingInfo
+     * @param installationDir installation directory of the current server instance.
+     * @return a ModelNode representing the instance product and the plateform it is running on.
+     * @throws OperationFailedException
+     */
+    protected ModelNode createProductNode(OperationContext context, ProductEnvironment environment,
+            ModelNode patchingInfo, Path installationDir) throws OperationFailedException {
+        assert environment != null;
+        ModelNode product = new ModelNode().setEmptyObject();
+        product.get(REPORT_VERSION).set("1.0");
+        product.get(NODE_NAME).set(environment.getHostName());
+        product.get(HOSTNAME).set(environment.getQualifiedHostName());
+        product.get(INSTANCE_ID).set(environment.getInstanceUuid().toString());
+        PathAddress organizationAddress = PathAddress.EMPTY_ADDRESS;
+        if(context.getProcessType() == ProcessType.HOST_CONTROLLER) {
+            organizationAddress = PathAddress.pathAddress(HOST, environment.getHostControllerName());
+        }
+        ModelNode root = context.readResourceFromRoot(organizationAddress, false).getModel();
+        if(root.hasDefined(ORGANIZATION)) {
+            product.get(ORGANIZATION).set(root.get(ORGANIZATION));
+        }
+        if (environment.getProductConfig() != null) {
+            if(environment.getProductConfig().getProductName() != null) {
+                product.get(PRODUCT_NAME).set(environment.getProductConfig().getProductName());
+            }
+            if(environment.getProductConfig().getProductVersion() != null) {
+                product.get(PRODUCT_VERSION).set(environment.getProductConfig().getProductVersion());
+            }
+            if (environment.getProductConfig().isProduct()) {
+                product.get(PRODUCT_COMMUNITY_IDENTIFIER).set(PRODUCT_TYPE);
+            } else {
+                product.get(PRODUCT_COMMUNITY_IDENTIFIER).set(PROJECT_TYPE);
+            }
+        }
+        if(installationDir != null && Files.exists(installationDir)) {
```

Comment:
This needs to be dealt with in the callers to this method, but the wildfly-swarm case needs to be handled. See ServerEnvironment L342 for what I'm getting at:

```

        if ( launchType.equals( LaunchType.SELF_CONTAINED ) ) {
            homeDir = new File(WildFlySecurityManager.getPropertyPrivileged("user.dir", "."));
```

With LaunchType.SELF_CONTAINED (also visible as ProcessType.SELF_CONTAINED) there really is no meaningful installation dir.